### PR TITLE
feat(builder): support multiple flashbots builders

### LINKED ISSUE #283: builder: Send bundle to more builders on L1
**Describe the feature**
https://docs.flashbots.net/flashbots-protect/rpc/mev-share#builders

Add a CLI option to modify the builders that receive bundles. Update transaction status tracking to handle multiple builders.

### PR DESCRIPTION
**Pull request by dancoombs**
Closes #283

### CODE REVIEW

File: crates/builder/src/sender/mod.rs [hunk 1]
```diff
@@ -82,100 +79,100 @@ pub(crate) trait TransactionSender: Send + Sync + 'static {
...
-                    return Err(SenderConstructorErrors::InvalidChainForSender(
-                        chain_spec.id,
-                        self.into_snake_case(),
-                    ));
-                }
-                TransactionSenderEnum::Flashbots(FlashbotsTransactionSender::new(client, signer)?)
+            Self::Flashbots(args) => {
+                let flashbots_signer = args
+                    .auth_key
+                    .parse::<LocalWallet>()
```
Comment (0xfourzerofour) [inline crates/builder/src/sender/mod.rs:157]: Nit: I think you do not need to explicitly define the type as it should be inferred by the input type of the function

**Review by 0xfourzerofour — APPROVED** [review]
Looks good just a tiny nit

File: crates/builder/src/sender/flashbots.rs [hunk 2]
```diff
@@ -96,38 +97,97 @@ where
...
+struct Preferences {
+    fast: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    privacy: Option<OurPrivacy>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    validity: Option<Validity>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct OurPrivacy {
```
Comment (alex-miao) [inline crates/builder/src/sender/flashbots.rs:163]: nit: why is this called `OurPrivacy` instead of just `Privacy`

**Review by alex-miao — APPROVED** [review]
Awesome, nice work figuring out the auth issues I was having with the Flashbots endpoint before

File: crates/builder/src/sender/mod.rs (hunk 1 above)
Comment (dancoombs) [reply crates/builder/src/sender/mod.rs:157]: Fixed

File: crates/builder/src/sender/flashbots.rs (hunk 2 above)
Comment (dancoombs) [reply crates/builder/src/sender/flashbots.rs:163]: Good catch, left over from some testing I was doing

**Review by 0xfourzerofour — APPROVED** [review]

**Review by dphilipson — APPROVED** [review]

**Pull request merged**